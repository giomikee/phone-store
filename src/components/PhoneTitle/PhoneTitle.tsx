import type { Phone } from '../../interfaces';
import { Rating, Skeleton, Stack, Typography } from '@mui/material';
import PhoneChips from '../PhoneChips/PhoneChips';
import { HEADINGS } from '../../constants';

const PhoneTitle = ({ phoneDetails }: { phoneDetails: Phone | null }) => {

    if (!phoneDetails) {
        return (
            <Stack spacing={1}>
                <Skeleton width={150} />
                <Skeleton width={150} />
                <Skeleton width={200} />
            </Stack>
        )
    }

    return (
        <Stack spacing={1}>
            <PhoneChips 
                basePrice={phoneDetails.basePrice} 
                brand={phoneDetails.brand}
                brandChipStyle={{ mr: '4px' }} /> 
            
            <Stack direction='row' spacing={1}>
                <Rating
                    defaultValue={phoneDetails.rating}
                    value={phoneDetails.rating}
                    precision={0.5}
                    readOnly 
                />

                <Typography component='span'>
                    {phoneDetails.rating} de 5 estrellas
                </Typography>
            </Stack>
            
            <Typography 
                component={HEADINGS.h2}
                variant={HEADINGS.h5}
            >
                {phoneDetails.name}
            </Typography>
        </Stack> 
    )
}

export default PhoneTitle