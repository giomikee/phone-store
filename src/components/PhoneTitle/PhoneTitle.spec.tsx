import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import PhoneTitle from './PhoneTitle';

describe('Given PhoneTitle component', () => { 
    const defaultProps = {
        phoneDetails: {
            id: '1',
            brand: 'Apple',
            name: 'iPhone 15',
            basePrice: 999,
            rating: 4,
        },
    }

    describe.each([
        defaultProps.phoneDetails,
        null,
    ])('When component is rendered', (phoneDetails) => {
        describe(`And phoneDetails is ${phoneDetails ? 'not null' : 'null'}`, () => { 
            it('Then it should match the snapshot', () => {
                const { container } = render(<PhoneTitle {...defaultProps} phoneDetails={phoneDetails} />);
                expect(container).toMatchSnapshot();
            });
        });
    });
 })