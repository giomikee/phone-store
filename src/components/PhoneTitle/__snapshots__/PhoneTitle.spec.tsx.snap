// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given PhoneTitle component > When component is rendered > And phoneDetails is not null > Then it should match the snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-jfdv4h-MuiStack-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary css-12mrdyl-MuiChip-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorPrimary css-1umw9bq-MuiSvgIcon-root"
          data-testid="SmartphoneIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17 1.01 7 1c-1.1 0-2 .9-2 2v18c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V3c0-1.1-.9-1.99-2-1.99M17 19H7V5h10z"
          />
        </svg>
        <span
          class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
        >
          Apple
        </span>
      </div>
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSecondary MuiChip-filledSecondary css-3p8epo-MuiChip-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorSecondary css-1umw9bq-MuiSvgIcon-root"
          data-testid="EuroIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5H15l1-2H8.58c-.05-.33-.08-.66-.08-1s.03-.67.08-1H15l1-2H9.24C10.32 6.92 12.5 5.5 15 5.5c1.61 0 3.09.59 4.23 1.57L21 5.3C19.41 3.87 17.3 3 15 3c-3.92 0-7.24 2.51-8.48 6H3l-1 2h4.06c-.04.33-.06.66-.06 1s.02.67.06 1H3l-1 2h4.52c1.24 3.49 4.56 6 8.48 6 2.31 0 4.41-.87 6-2.3l-1.78-1.77c-1.13.98-2.6 1.57-4.22 1.57"
          />
        </svg>
        <span
          class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
        >
          desde 999.00 â‚¬
        </span>
      </div>
    </div>
    <div
      class="MuiStack-root css-niqf4j-MuiStack-root"
    >
      <span
        aria-label="4 Stars"
        class="MuiRating-root MuiRating-sizeMedium Mui-readOnly css-i0l71t-MuiRating-root"
        role="img"
      >
        <span
          class="MuiRating-decimal css-1czqjfa-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="MuiRating-decimal css-1czqjfa-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="MuiRating-decimal css-1czqjfa-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="MuiRating-decimal css-1czqjfa-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="MuiRating-decimal css-1czqjfa-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconEmpty css-9xw0na-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                data-testid="StarBorderIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconEmpty css-9xw0na-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                data-testid="StarBorderIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                />
              </svg>
            </span>
          </span>
        </span>
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
      >
        4
         de 5 estrellas
      </span>
    </div>
    <h2
      class="MuiTypography-root MuiTypography-h5 css-16xl4zq-MuiTypography-root"
    >
      iPhone 15
    </h2>
  </div>
</div>
`;

exports[`Given PhoneTitle component > When component is rendered > And phoneDetails is null > Then it should match the snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-jfdv4h-MuiStack-root"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
      style="width: 150px;"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
      style="width: 150px;"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
      style="width: 200px;"
    />
  </div>
</div>
`;
